### PR TITLE
fix json emitter for empty response bodies

### PR DESCRIPTION
**Pull request by demianbrecht**
- [ ] Update CHANGELOG.md
- [ ] Update README.md (as needed)
- [ ] New dependencies were added to `pyproject.toml`
- [ ] `pip install` succeeds with a clean virtualenv
- [ ] There are new or modified tests that cover changes
- [ ] Test coverage is maintained or expanded

### CODE REVIEW

**Comment by bgrant** [comment]
Can you add a CHANGELOG entry?

File: tests/resources/test_emitters.py [hunk 1]
```diff
@@ -109,6 +109,11 @@ def test_decode(self):
         resp = em.render(django.test.RequestFactory().get("/"))
         self.assertEqual(json.loads(resp), ["foo", "bar"])
 
+    def test_decode_empty_list(self):
+        em = emitters.JSONEmitter([""], lambda: None)
```
Comment (lukka5) [inline tests/resources/test_emitters.py:113]: for completeness don't we want another test with a truly empty list?

File: django_declarative_apis/resources/emitters.py [hunk 2]
```diff
@@ -183,19 +183,23 @@ def decode(self, data):
...
         assert cb is None, "JSONP Callbacks not suppoted"
-        seria = json.dumps(
-            self.decode(self.construct()),
-            cls=DjangoJSONEncoder,
-            ensure_ascii=False,
-            indent=4,
-        )
+        seria = self.decode(self.construct())
+        if isinstance(seria, list):
+            if len(seria) == 0 or (len(seria) == 1 and len(seria[0]) == 0):
```
[the remark this answers is not in the document]
Comment (demianbrecht) [reply django_declarative_apis/resources/emitters.py:188]: The base case (len == 0) is a just in case. What we actually see with our current issue is `[""]`. No, if we have a list of length 1, it would always be accessible and have a length given it's a string (`val.decode("utf8")` in self.decode)

**Review by bgrant — DISMISSED** [review]

**Review by lukka5 — DISMISSED** [review]

**Review by bgrant — APPROVED** [review]
